### PR TITLE
#9 Initial implementation of generate command

### PR DESCRIPTION
**Pull request by antoniaklja**
#9 Initial implementation of generate command

### CODE REVIEW

File: generate/generate.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	return privateKey, nil
+}
+
+func generatePublicKey(privateKey *rsa.PrivateKey) *rsa.PublicKey {
+	publicKey := &privateKey.PublicKey
+	return publicKey
+}
+
+func savePrivateKey(privateKey *rsa.PrivateKey) error {
+	PrivASN1 := x509.MarshalPKCS1PrivateKey(privateKey)
```
Comment (pawelprazak) [inline generate/generate.go:29]: what's the relation between `ASN1` and `PKCS1`, I'm not familiar with either :thinking:

File: generate/generate.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+	privBytes := pem.EncodeToMemory(&pem.Block{
+		Type:  "RSA PRIVATE KEY",
+		Bytes: PrivASN1,
+	})
+	ioutil.WriteFile("private.key", privBytes, 0644) // FIXME @antoniaklja shall we pass destination path here?
```
Comment (pawelprazak) [inline generate/generate.go:35]: shouldn't the access be `600` or even `400`?

File: main.go [hunk 3]
```diff
@@ -18,6 +19,7 @@ func main() {
       Usage:   "generate a key and certificate pair",
       Action:  func(c *cli.Context) error {
         fmt.Println("generate is not implemented yet")
```
Comment (pawelprazak) [inline main.go:21]: the message looks no longer true

File: generate/generate.go (hunk 1 above)
Comment (antoniaklja) [reply generate/generate.go:29]: This is the common way how to marshall key using rsa golang package.
I found following description: 

`in cryptography, PKCS #1 is the first of a family of standards called Public-Key Cryptography Standards (PKCS), published by RSA Laboratories. It provides the basic definitions of and recommendations for implementing the RSA algorithm for public-key cryptography. It defines the mathematical properties of public and private keys, primitive operations for encryption and signatures, secure cryptographic schemes, and related ASN.1 syntax representations.`

also take a look here https://tls.mbed.org/kb/cryptography/asn1-key-structures-in-der-and-pem

> in essence PEM files are just base64 encoded versions of the DER encoded data

File: generate/generate.go (hunk 2 above)
Comment (antoniaklja) [reply generate/generate.go:35]: let's use 600 - owner read write

File: generate/generate.go (hunk 1 above)
Comment (pawelprazak) [reply generate/generate.go:29]: thanks!

**Pull request merged**